src/Optimizely/ProjectConfig.php

```diff
@@ -150,12 +178,16 @@ public function __construct($datafile, $logger, $errorHandler)
         $events = $config['events'] ?: [];
         $attributes = $config['attributes'] ?: [];
         $audiences = $config['audiences'] ?: [];
+        $rollouts = isset($config['rollouts']) ? $config['rollouts'] : [];
+        $featureFlags = isset($config['featureFlags']) ? $config['featureFlags']: [];
 
         $this->_groupIdMap = ConfigParser::generateMap($groups, 'id', Group::class);
         $this->_experimentKeyMap = ConfigParser::generateMap($experiments, 'key', Experiment::class);
         $this->_eventKeyMap = ConfigParser::generateMap($events, 'key', Event::class);
         $this->_attributeKeyMap = ConfigParser::generateMap($attributes, 'key', Attribute::class);
         $this->_audienceIdMap = ConfigParser::generateMap($audiences, 'id', Audience::class);
+        $this->_rollouts = ConfigParser::generateMap($rollouts, null, Rollout::class);
```

Comment:
why we are generating map? if we are not passing key or id other than null?

Reply:
When given null as 2nd param, ConfigParser::generateMap makes a list